Ticket#6893 Agrego seguimiento de handles de item, colecciones y comunidades ante cada pageView de un item en GoogleAnalytics

santit96 opened the pull request:
Agrego codigo para que se registren a que item, colecciones y comunidades se les está haciendo un hit en el caso de una visita a la página de un ítem. Para ello creo 2 dimensiones personalizadas, una para items y otra para colecciones y comunidades

Los handles de las comunidades y colecciones del item se envian separados por coma en la misma dimensión. Para obtenerlos desde el xsl se creó el método getItemCollsAndComms en la clase XslExtensions.

santit96 committed:
-#6893 Modifico el código de la etiqueta de google analytics para adecuarlo a lo que dice la documentacion
	Agrego codigo para que se registren a que item y colección se le está haciendo un hit en el caso de una visita a la página de un ítem. Para ello creo 2 dimensiones personalizadas, una para items y otra para colecciones

santit96 committed:
-#6893 En el envio de datos a google analytics ahora se envian como una sola dimension todos los handles de las colecciones y comunidades de un item separados por coma
	Antes solo se enviaban algunos, los que estaban en el DRI, ahora se llama a un metodo que los trae a todos
	Luego se envian en una misma dimension

santit96 committed:
-#6893 Fix en el script js de Google analytic
	Agrego ; faltantes
	Cambio el nombre del metodo que obtiene un listado de comunidades y colecciones de un item por getItemCollsAndComms

arieljlira commented on dspace/modules/xmlui/src/main/webapp/themes/Sedici2/lib/xsl/core/page-structure.xsl:
```diff
@@ -614,17 +615,25 @@ placeholders for header images -->
+				</xsl:text>
+				<xsl:if test="/dri:document/dri:meta/dri:pageMeta/dri:metadata[@element='focus'][@qualifier='containerType']/text() = 'type:item'">
+					<xsl:variable name="itemHandle" select="substring(/dri:document/dri:meta/dri:pageMeta/dri:metadata[@element='focus'][@qualifier='object']/text(), 5)"/><xsl:text>
+					ga('set', 'dimension1', '</xsl:text><xsl:value-of select="$itemHandle"/><xsl:text>');
+					ga('set', 'dimension2', '</xsl:text><xsl:value-of select="ex:getItemCollsAndComms($itemHandle)"/><xsl:text>');
```
Sería bueno ponerle dimensiones con nombres autoexplicativos, se puede?

santit96 replied:
Desde el código no, hay que referirse a las dimensiones por el índice (dimension[dimensionIndex]). Para ver los nombres de las dimensiones hay que ir a la UI de analytics

arieljlira commented on dspace/modules/xmlui/src/main/webapp/themes/Sedici2/lib/xsl/core/page-structure.xsl:
```diff
@@ -614,17 +615,25 @@ placeholders for header images -->
+
+				ga('create', '</xsl:text><xsl:value-of select="/dri:document/dri:meta/dri:pageMeta/dri:metadata[@element='google'][@qualifier='analytics']"/><xsl:text>', 'auto');
+				</xsl:text>
+				<xsl:if test="/dri:document/dri:meta/dri:pageMeta/dri:metadata[@element='focus'][@qualifier='containerType']/text() = 'type:item'">
```
Sólo se agregan los datos adicionales cuando se esta viendo un item? Quizas se puede ampliar para que siempre se registre la info contextual que se puede. Menciono algunos ejemplos sin saber si es conveniente o costoso: 
- dso=handle del item, col, o comm que se esta viendo
- containers/parents (cols and comms)
- itemType=dc.type 
- etc

santit96 replied:
Si, se podría tranquilamente, solo lo había puesto para items porque no sabía si también se quería registrar esto para las paginas de comunidades y colecciones

arieljlira reviewed (commented):
Lo veo muy bien @santit96  , puse un par de comentarios nomás.

santit96 committed:
=#6893 Agrego otra dimensión a los datos enviados a Google Analytics que registra ante cada pageView de un item el tipo de item (articulo, tesis, libro, etc)

FacundoAdorno reviewed (approved):
El código me parece bien. También lo probé y manda los datos a Analytics. Por ahí también estaría bueno pensar en agregar el subtype (o combinar "Type->Subtype" o algo asi).

santit96 committed:
-#6893 Ahora tambien se envian las dimensiones personalizadas de google analytics en el pageView de una coleccion y de una comunidad
	La dimension1 pasó a ser el handle del DSO al que se la hizo el page view
	La dimension2 es el listados de handles de los objetos padres (comunidades y colecciones) del DSO
	La dimension3 sigue siendo el tipo de item en el caso del pageview de un item (articulo, libro, etc)
	Se agregaron 3 nuevas dimensiones:\
	La dimension 4 es el tipo de DSO al que se le hizo el pageview (item, coleccion o comunidad)
	La dimension 5 es el subtipo del item
	La dimension 6 es la fecha de publicacion (date issued) del item

santit96 committed:
-#6893 Fix al codigo js de google analytics, corrijo el nombre de una variable que estaba mal

santit96 commented:
Para ver si funciona en analytics posta pueden cambiar la propiedad xmlui.google.analytics.key del dspace.cfg seteandola con el ID de seguimiento del anaytics de SEDICI

FacundoAdorno reviewed (approved):
Todas las dimensiones se estan enviando bien. No se muy bien como maneja Analytics la parte de las fechas, si usa algun formato estandar ISO o algo asi, igualmente, con los datos enviados se puede filtrar usando expresiones regulares. Por ejemplo, si se quieren todos los items superiores al 2015, se podria utilizar la siguiente expresion: `201[6-9]|20[2-9]\d`.
